.github/workflows/performance.yml

```diff
@@ -36,5 +38,23 @@ jobs:
       run: |
         npm ci
 
-    - name: Run the performance tests
+    - name: Compare performance with master
+      if: github.event_name == 'pull_request'
       run: ./bin/plugin/cli.js perf --ci $GITHUB_SHA master --tests-branch $GITHUB_SHA
+
+    - name: Compare performance with current WordPress Core and previous Gutenberg versions
+      if: github.event_name == 'release'
+      env:
+        PLUGIN_VERSION: ${{ github.event.release.name }}
+      run: |
+        IFS='.' read -r -a PLUGIN_VERSION_ARRAY <<< "$PLUGIN_VERSION"
+        CURRENT_RELEASE_BRANCH="release/${PLUGIN_VERSION_ARRAY[0]}.${PLUGIN_VERSION_ARRAY[1]}"
+        PREVIOUS_VERSION_BASE_10=$(expr ${PLUGIN_VERSION_ARRAY[0]} \* 10 + ${PLUGIN_VERSION_ARRAY[1]} - 1)
+        PREVIOUS_RELEASE_BRANCH="release/$(expr $PREVIOUS_VERSION_BASE_10 / 10).$(expr $PREVIOUS_VERSION_BASE_10 % 10)"
+        TESTED_UP_TO_REGEX="Tested up to: \K([0-9]+)\.([0-9]+)\.?([0-9]?)"
+        WP_VERSION=$(grep -oP "$TESTED_UP_TO_REGEX" ./readme.txt)
+        IFS='.' read -r -a WP_VERSION_ARRAY <<< "$WP_VERSION"
+        WP_BRANCH="wp/${WP_VERSION_ARRAY[0]}.${WP_VERSION_ARRAY[1]}"
+        ./bin/plugin/cli.js perf --ci $WP_BRANCH $PREVIOUS_RELEASE_BRANCH $CURRENT_RELEASE_BRANCH
```

Comment:
This `run` block uses Bash-specific constructs (`<<<` syntax, arrays), so it doesn't qualify as shell-agnostic. Being unfamiliar with GitHub Actions, I looked at the [documentation](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell) to understand how shell defaults work. It's a little ambiguous, but then you see:

> `jobs.<job_id>.steps[*].run` — Runs command-line programs using the operating system's shell.

So I guess our runners default to Linux, and so the Bash shell is assumed. But perhaps we could make it explicit?

```yaml
shell: bash
run: |
    IFS='.' read ...
```

@ockham: wanted your thoughts before I do anything.

***

The second thing that I noticed is that this block is using `grep -P` (`--perl-regexp`), which is a GNU extension and not POSIX. It's not a major issue, since our runners are likely always GNU/Linux, but it _would be nice_ to be POSIX, I guess. (I'm more likely to run into these issues than you, since macOS is BSD-ish. 😄)

Reply:
Yeah, definitely using a few bashisms here 😬  I was choosing the path of least resistance in a few cases, which seemed to be offered by Bash.

I'm pretty sure GH Action runners are Linux (Ubuntu LTS) by default, and those Ubuntu shells default to bash. I don't think GH would change that without prior notice, so I don't think that setting the `shell` explicitly is a strict requirement, but I also don't mind having it.


***

About the `grep -P` GNU extension -- I ran into this the other day when working on [another workflow](https://github.com/WordPress/gutenberg/pull/31897), now that I'm on macOS myself (and ended up replacing it by something POSIX compliant). Both this and the Bashism above show that I originally wrote these on Linux. I'm fine replacing them by something more cross-platform, since it'll make it easier for people to test these locally.